test_views.py

```
@@ -0,0 +1,209 @@
+        product.save()  # Individual saves instead of bulk_update
+    return HttpResponse("Prices updated")
+
+# Security issue - no rate limiting
+def api_endpoint(request):
+    # No rate limiting protection
+    data = request.POST.get('data')
+    # Process data...
+    return JsonResponse({'result': 'success'})
+
+# Bad practice - no proper HTTP status codes
+def delete_product(request, product_id):
```

Ruff (0.12.2)

Unused function argument: `request`

(ARG001)